sdk/log/logtest/builder.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package logtest is a testing helper package.
+package logtest // import "go.opentelemetry.io/otel/sdk/log/logtest"
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// RecordFactory is used to facilitate unit testing implementations of
+// [go.opentelemetry.io/otel/sdk/log.Exporter]
+// and [go.opentelemetry.io/otel/sdk/log.Processor].
+//
+// Do not use RecordFactory to create records in production code.
+type RecordFactory struct {
+	Timestamp         time.Time
+	ObservedTimestamp time.Time
+	Severity          log.Severity
+	SeverityText      string
+	Body              log.Value
+	Attributes        []log.KeyValue
+	TraceID           trace.TraceID
+	SpanID            trace.SpanID
+	TraceFlags        trace.TraceFlags
+
+	Resource             *resource.Resource
+	InstrumentationScope instrumentation.Scope
+
+	DroppedAttributes int
+}
+
+// Record returns a log record.
+func (b RecordFactory) Record() sdklog.Record {
```

Comment:
nit: `NewRecord` might convey the creation functionality a bit better.

```suggestion
func (b RecordFactory) NewRecord() sdklog.Record {
```

Not critical as this looks good enough to start with.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/5258/commits/2bb62ff22f29e2894b3f8aea0e186c4084b3e7a6